pxr/imaging/hdSt/CMakeLists.txt

```diff
@@ -626,6 +617,28 @@ pxr_register_test(testHdStQualifiers
         TF_DEBUG=HD_SAFE_MODE
 )
 
+if (${PXR_ENABLE_MATERIALX_SUPPORT})
+pxr_build_test(testHdStMaterialXShaderGen
+    LIBRARIES
+        hdSt
+        hdMtlx
+        tf
+    CPPFILES
+        testenv/testHdStMaterialXShaderGen.cpp
+)
+pxr_install_test_dir(
+    SRC testenv/testHdStMaterialXShaderGen
+    DEST testHdStMaterialXShaderGen
+)
+pxr_register_test(testHdStMaterialXShaderGen_testCacheSerialization
+    COMMAND "${CMAKE_INSTALL_PREFIX}/tests/testHdStMaterialXShaderGen --testCacheSerialization"
+    EXPECTED_RETURN_CODE 0
+    STDOUT_REDIRECT shadergen_testCacheSerialization.out
+    DIFF_COMPARE shadergen_testCacheSerialization.out
+    TESTENV testHdStMaterialXShaderGen
+)
```

Comment:
The new test I added that tests if metadata is serialized correctly by the new MaterialX codegen cache. It reuses an existing test executable `testHdStMaterialXShaderGen` with a new command-line argument.